specification/guestconfiguration/GuestConfiguration.Management/models.tsp

```diff
@@ -0,0 +1,801 @@
+  complianceStatus?: ComplianceStatus;
+
+  /**
+   * Date and time when last compliance status was checked.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  lastComplianceStatusChecked?: utcDateTime | null;
+
+  /**
+   * Id of the latest report for the guest configuration assignment.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  latestReportId?: string | null;
+
+  /**
+   * parameter hash for the guest configuration assignment.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  parameterHash?: string | null;
+
+  /**
+   * Last reported guest configuration assignment report.
+   */
+  latestAssignmentReport?: AssignmentReport;
+
+  /**
+   * The source which initiated the guest configuration assignment. Ex: Azure Policy
+   */
+  context?: string;
+
+  /**
+   * Combined hash of the configuration package and parameters.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  assignmentHash?: string | null;
+
+  /**
+   * The provisioning state, which only appears in the response.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState | null;
+
+  /**
+   * Type of the resource - VMSS / VM
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  resourceType?: string | null;
+
+  /**
+   * The list of VM Compliance data for VMSS
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @OpenAPI.extension("x-ms-identifiers", #["vmId"])
+  vmssVMList?: VmssvmInfo[];
+}
+
+/**
+ * Guest configuration is an artifact that encapsulates DSC configuration and its dependencies. The artifact is a zip file containing DSC configuration (as MOF) and dependent resources and other dependencies like modules.
+ */
+model GuestConfigurationNavigation {
+  /**
+   * Kind of the guest configuration. For example:DSC
+   */
+  kind?: Kind;
+
+  /**
+   * Name of the guest configuration.
+   */
+  name?: string;
+
+  /**
+   * Version of the guest configuration.
+   */
+  version?: string;
+
+  /**
+   * Uri of the storage where guest configuration package is uploaded.
+   */
+  contentUri?: string;
+
+  /**
+   * Combined hash of the guest configuration package and configuration parameters.
+   */
```

Comment:
whats the hash algorithm?

If its SHA1 CryptoBoard is not happy, as its considered insecure, there needs to be an alternative to use secure hash algorithms.